CONTRIBUTING.md

```diff
@@ -0,0 +1,187 @@
+* [Coverage](https://github.com/JuliaCI/Coverage.jl) can be useful for checking if you've lowered the overall test coverage against [master](https://coveralls.io/r/JuliaStats/StatsBase.jl?branch=master)
+
+## Style
+
+### Summary
+
+* Use 4 spaces per indentation level, no tabs.
+* Try to adhere to the 92 character line length limit.
+* Use upper camel case convention for [modules](http://julia.readthedocs.org/en/latest/manual/modules/) and [types](http://julia.readthedocs.org/en/latest/manual/types/).
+* Use lower case for method names, preferably without underscores.
+* Comments are a good way to explain the intentions of your code and can aid in Pull Request (PR) reviews.
+* No whitespace at the end of a line (trailing whitespace).
+* Avoid padding brackets with spaces. ex. `Int64(value)` preferred over `Int64( value )`.
+* Include spaces after commas and around operators
+
+### Methods Definitions
+
+Short form method definitions when the method body only contains 1 statement:
+
+```julia
+# Yes
+Base.stdm(v::RealArray, w::AbstractWeights, m::Real; corrected::DepBool=nothing) =
+    sqrt(varm(v, w, m, corrected=depcheck(:stdm, corrected)))
+
+# No
+function Base.stdm(v::RealArray, w::AbstractWeights, m::Real; corrected::DepBool=nothing)
+    sqrt(varm(v, w, m, corrected=depcheck(:stdm, corrected)))
+end
+```
+
+For long method definitions (and calls) our preferences is to align arguments on new lines with the first argument and avoid adding an extra newline after the last argument:
+
+```julia
+# Yes
+function Base.var(v::RealArray, w::AbstractWeights; mean=nothing,
+                  corrected::DepBool=nothing)
+    ...
+end
+
+# No
+function Base.var(
+    v::RealArray, w::AbstractWeights; mean=nothing, corrected::DepBool=nothing
+)
+    ...
+end
+
+# No
+function Base.var(
+    v::RealArray,
+    w::AbstractWeights;
+    mean=nothing,
+    corrected::DepBool=nothing
+)
+    ...
+end
+
+# Yes
+scale!(_wsum_centralize!(R, abs2, A, values(w), M, dim, true),
+       varcorrection(w, corrected))
+
+# No
+scale!(
+    _wsum_centralize!(R, abs2, A, values(w), M, dim, true),
+    varcorrection(w, corrected)
+)
+```
+
+### `return`
+
+For simple methods, we try to avoid using the `return` keyword:
+
+```julia
+# Yes
+function Base.var(v::RealArray, w::AbstractWeights; mean=nothing,
+                  corrected::DepBool=nothing)
+    corrected = depcheck(:var, corrected)
+
+    if mean == nothing
+        varm(v, w, Base.mean(v, w); corrected=corrected)
+    else
+        varm(v, w, mean; corrected=corrected)
+    end
+end
+
+# No
+function Base.var(v::RealArray, w::AbstractWeights; mean=nothing,
+                  corrected::DepBool=nothing)
+    corrected = depcheck(:var, corrected)
+
+    if mean == nothing
+        return varm(v, w, Base.mean(v, w); corrected=corrected)
+    else
+        return varm(v, w, mean; corrected=corrected)
+    end
+end
+```
+
+However, for more complicated methods it's helpful to be more explicit:
```

Comment:
Honestly, I'm not sure everybody would agree on this one as it's pretty clear that the last line returns here. I wouldn't give this example.